modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/SessionWindowProcessor.java

```diff
@@ -0,0 +1,573 @@
+        }
+
+        for (Map.Entry<String, Long> entry : previousEndTimestamps.entrySet()) {
+            previousSession = sessionMap.get(entry.getKey()).getPreviousSession();
+
+            if (previousSession != null && previousSession.getFirst() != null &&
+                    eventTimestamp >= previousSession.getAliveTimestamp()) {
+
+                expiredEventChunk.setKey(previousSession.getKey());
+                expiredEventChunk.setTimestamps(previousSession.getStartTimestamp(),
+                        previousSession.getEndTimestamp(), previousSession.getAliveTimestamp());
+
+                expiredEventChunk.add(previousSession.getFirst());
+                previousSession.clear();
+            } else {
+                break;
+            }
+
+        }
+
+    }
+
+    /**
+     * Gets all end timestamps of current sessions.
+     * @param sessionMap holds all the sessions with the session key
+     * @return map with the values of each current session's end timestamp and with the key as the session key
+     */
+    private Map<String, Long> findAllCurrentEndTimestamps(Map<String, SessionContainer> sessionMap) {
+
+        Collection<SessionContainer> sessionContainerList =  sessionMap.values();
+
+        if (!sessionKeyEndTimeMap.isEmpty()) {
+            sessionKeyEndTimeMap.clear();
+        }
+
+        for (SessionContainer sessionContainer : sessionContainerList) {
+            //not getting empty session details
+            if (sessionContainer.getCurrentSessionEndTimestamp() != -1) {
+                sessionKeyEndTimeMap.put(sessionContainer.getKey(), sessionContainer.getCurrentSessionEndTimestamp());
+            }
+        }
+
+        return sessionKeyEndTimeMap;
+    }
+
+    /**
+     * Gets all the end timestamps of previous sessions.
+     * @return map with the values of each previous session's end timestamp and with the key as the sesssio key
+     */
+    private Map<String, Long> findAllPreviousEndTimestamps(Map<String, SessionContainer> sessionMap) {
+
+        Collection<SessionContainer> sessionContainerList =  sessionMap.values();
+
+        if (!sessionKeyEndTimeMap.isEmpty()) {
+            sessionKeyEndTimeMap.clear();
+        }
+
+        for (SessionContainer sessionContainer : sessionContainerList) {
+            //not getting empty session details
+            if (sessionContainer.getPreviousSessionEndTimestamp() != -1) {
+                sessionKeyEndTimeMap.put(sessionContainer.getKey(), sessionContainer.getPreviousSessionEndTimestamp());
+            }
+        }
+
+        return sessionKeyEndTimeMap;
+    }
+
+    @Override
+    public void start() {
+        //Do nothing
+    }
+
+    @Override
+    public void stop() {
+        //Do nothing
+    }
+
+    @Override
+    public synchronized Map<String, Object> currentState() {
+        Map<String, Object> state = new HashMap<>();
+        state.put("sessionMap", sessionMap);
+        state.put("sessionGap", sessionGap);
```

Comment:
You dont need to persist sessionGap and allowedLatency as they are input parameters

Reply:
done accordingly